helpers/utils.py

```diff
@@ -107,4 +128,117 @@ def UmaskNamedTemporaryFile(*args, **kargs):
     umask = os.umask(0o666)
     os.umask(umask)
     os.chmod(fdesc.name, 0o666 & ~umask)
-    return fdesc
\ No newline at end of file
+    return fdesc
+
+def build_kubectl_cmd(namespace: str, k8s_context: str) -> str:
+    kubeconfig = get_kube_config()
+    kubectl_cmd = get_kubectl_cmd()
+    cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig} --namespace={namespace}"
+    if k8s_context:
+        cmd += f" --context {k8s_context}"
+    return cmd
+
+def build_secret_name(username: str, namespace: str) -> str:
+    return f"spark-client-sa-conf-{username or 'spark'}"
+
+def execute_kubectl_cmd(cmd: str, exit_code_on_error: int) -> str:
+    logging.debug(cmd)
+    # kubeconfig = get_kube_config()
+    try:
+        out = subprocess.check_output(cmd, shell=True)
+        result = out.decode('utf-8') if out else None
+    except subprocess.CalledProcessError as e:
+        # logging.error(
+        #     f'Probably bad or missing kubeconfig file {kubeconfig}')
+        # logging.warning(
+        #     f'Please set the KUBECONFIG environment variable to point to the kubeconfig. Or fix the default kubeconfig {kubeconfig}')
+        logging.error(e.output)
+        sys.exit(exit_code_on_error)
+
+    return result
+
+def setup_kubernetes_secret_literal(username: str, namespace: str, k8s_context: str, conf : List[str]) -> None:
+    kubectl_cmd = build_kubectl_cmd(namespace, k8s_context)
+    secret_name = build_secret_name(username, namespace)
+    cmd = f"{kubectl_cmd} create secret generic {secret_name}"
+
+    if conf:
+        for c in conf:
+            cmd += f" --from-literal={c}"
+
+    defaults = generate_spark_default_conf()
+    for k in defaults.keys():
+        cmd += f" --from-literal={k}={defaults[k]}"
+
+    cmd += f" --from-literal=spark.kubernetes.authenticate.driver.serviceAccountName={username}"
+    cmd += f" --from-literal=spark.kubernetes.namespace={namespace}"
```

Comment:
No need to create these. These will be automatically generated and merged in spark-submit.